libgraphics/include/pivot/graphics/types/Transform.hxx

```diff
@@ -13,62 +13,49 @@
 /// @brief Hold the model matrix
 class Transform
 {
-private:
-    struct DecomposedMatrix {
-        glm::vec3 scale;
-        glm::quat orientation;
-        glm::vec3 translation;
-        glm::vec3 skew;
-        glm::vec4 perspective;
-    };
-
 public:
     /// Default ctor
     Transform() = default;
 
     /// Constructor from vector
     inline Transform(const glm::vec3 &translation, const glm::vec3 &rotation, const glm::vec3 &scale)
-        : Transform(glm::translate(glm::mat4(1.0f), translation), glm::toMat4(glm::quat(rotation)),
-                    glm::scale(glm::mat4(1.0f), scale))
+        : position(translation), rotation(rotation), scale(scale)
     {
     }
 
-    /// Constructor from matrices
-    inline Transform(const glm::mat4 &translation, const glm::mat4 &rotation, const glm::mat4 &scale)
-        : m_matrix(decomposeMatrix(translation * rotation * scale))
+    /// Get the model matrix
+    inline glm::mat4 getModelMatrix() const noexcept { return recomposeMatrix(*this); }
+
+    /** \brief Transform comparison operator
+     *
+     * WARNING: The Transform cannot be ordered for now. They can only be
+     * compared equal or not.
+     */
+    std::partial_ordering operator<=>(const Transform &rhs) const
     {
+        if (position == rhs.position && rotation == rhs.rotation && scale == rhs.scale) {
+            return std::partial_ordering::equivalent;
+        }
+        return std::partial_ordering::unordered;
     }
```

Comment:
Why is the default `operator<=>` not available ? Is it because of the [GCC bug](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=104606) ?

Reply:
No it's because ordering transforms do not really make any sense. What would be a "bigger" transform ? The one with the largest volume ? The one furthest from the origin ? Which rotation is bigger ? The only reasonable comparison is equality.